operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/EventSourceAware.java

```diff
@@ -1,10 +1,36 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent;
 
+import java.util.Optional;
+
 import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.javaoperatorsdk.operator.processing.event.EventSourceRetriever;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.processing.event.source.ResourceEventSource;
+
+public interface EventSourceAware<R, P extends HasMetadata> {
+
+  /**
+   * Dependent resources are designed to by default provide event sources. There are cases where it
+   * might not:
+   * <ul>
+   * <li>If an event source is shared between multiple dependent resources. In this case only one or
+   * none of the dependent resources sharing the event source should provide one.</li>
+   * <li>Some special implementation of an event source. That just execute some action might not
+   * provide one.</li>
+   * </ul>
+   *
+   * @param context context of event source initialization
+   * @return an optional event source
+   */
+  default Optional<ResourceEventSource<R, P>> eventSource(EventSourceContext<P> context) {
+    return getUsedEventSourceName().map(
+        name -> context.getEventSourceRetriever().getResourceEventSourceFor(resourceType(), name));
+  }
 
-public interface EventSourceAware<P extends HasMetadata> {
+  Class<R> resourceType();
```

Comment:
It's a little smelly that we have resourceType in both places (also in dependent resource), but no big deal.

Reply:
Yeah, it's not ideal but this allows to have a default implementation in the interface…